HELP.md

```diff
@@ -0,0 +1,761 @@
+        }
+    }
+
+    pub fn set_status(&mut self, status: String) {
+        self.status_updates.insert(&env::predecessor_account_id(), &status);
+        // Note, don't need to check size, since `UnorderedMap` doesn't store all data in memory.
+    }
+
+    pub fn delete_status(&mut self) {
+        self.status_updates.remove(&env::predecessor_account_id());
+    }
+
+    pub fn get_status(&self, account_id: ValidAccountId) -> Option<String> {
+        self.status_updates.get(account_id.as_ref())
+    }
+}
+```
+
+## Pagination with persistent collections
+
+Persistent collections such as `UnorderedMap`, `UnorderedSet` and `Vector` may
+contain more elements than the amount of gas available to read them all.
+In order to expose them all through view calls, we can implement pagination.
+
+`Vector` returns elements by index natively using `.get(index)`.
+
+To access elements by index in `UnorderedSet` we can use `.as_vector()` that will return a `Vector` of elements.
+
+For `UnorderedMap` we need to get keys and values as `Vector` collections, using `.keys_as_vector()` and `.values_as_vector()` respectively.
+
+Example of pagination for `UnorderedMap`:
+
+```rust
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    pub status_updates: UnorderedMap<AccountId, String>,
+}
+
+#[near_bindgen]
+impl Contract {
+    /// Retrieves multiple elements from the `UnorderedMap`.
+    /// - `from_index` is the index to start from.
+    /// - `limit` is the maximum number of elements to return.
+    pub fn get_updates(&self, from_index: u64, limit: u64) -> Vec<(AccountId, String)> {
+        let keys = self.status_updates.keys_as_vector();
+        let values = self.status_updates.values_as_vector();
+        (from_index..std::cmp::min(from_index + limit, self.status_updates.len()))
+            .map(|index| (keys.get(index).unwrap(), values.get(index).unwrap()))
+            .collect()
+    }
+}
+```
+
+## `LookupMap` vs `UnorderedMap`
+
+### Functionality
+
+- `UnorderedMap` supports iteration over keys and values, and also supports pagination. Internally, it has the following structures:
+  - a map from a key to an index
+  - a vector of keys
+  - a vector of values
+- `LookupMap` only has a map from a key to a value. Without a vector of keys, it doesn't have the ability to iterate over keys.
+
+### Performance
+
+`LookupMap` has a better performance and stores less data comparing to the `UnorderedMap`.
+
+- `UnorderedMap` requires `2` storage reads to get the value and `3` storage writes to insert a new entry.
+- `LookupMap` requires only one storage read to get the value and only one storage write to store it.
+
+### Storage space
+
+`UnorderedMap` requires more storage for an entry comparing to a `LookupMap`.
+
+- `UnorderedMap` stores the key twice (once in the first map and once in the vector of keys) and value once. It also has higher constant for storing the length of vectors and prefixes.
```

Comment:
"higher constant"? Seems like maybe there's a missing word?

Reply:
Looking at the `fn insert_raw` from `LookupMap` and `UnorderedMap` seems like the latter has to incorportate `u64` storage. I'll add a word or two here for readability.